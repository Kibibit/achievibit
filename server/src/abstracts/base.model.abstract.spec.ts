import { MockModel } from '@kb-dev-tools';

describe('Base Model', () => {
  let model: MockModel;

  beforeEach(() => model = new MockModel({
    mockAttribute: 'nice',
    mockPrivateAttribute: 'bad'
  }));
  
  it ('should allow extending', () => {
    expect(model).toBeDefined();
  });

  it('should obfuscate and convert to plain object', () => {
    const asJson = model.toJSON();

    expect(model.mockAttribute).toBe('nice');
    expect(model.mockPrivateAttribute).toBe('bad');
    expect(asJson).toBeDefined();
    expect(asJson.mockPrivateAttribute).toBeUndefined();
    expect(asJson.mockAttribute).toBe('nice');
  });

  it('should obfuscate and convert to string', () => {
    const asString = model.toString();
    expect(asString).toMatchSnapshot();
    expect(asString).not.toMatch(/mockPrivateAttribute/g);
  });
});
