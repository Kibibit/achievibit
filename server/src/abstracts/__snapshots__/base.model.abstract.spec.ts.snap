// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Model should obfuscate and convert to string 1`] = `"{\\"mockAttribute\\":\\"nice\\"}"`;
